recipes/proj/all/conanfile.py

```diff
@@ -30,162 +38,153 @@ class ProjConan(ConanFile):
-            #       because we want absolute lib paths of build profile actually
-            if not hasattr(self, "settings_build"):
-                if tools.Version(self.version) < "8.1.0":
-                    cmake_sqlite_call = "CMakeLists.txt"
-                    pattern = "${EXE_SQLITE3}"
-                else:
-                    cmake_sqlite_call = "generate_proj_db.cmake"
-                    pattern = "\"${EXE_SQLITE3}\""
-                lib_paths = self.deps_cpp_info["sqlite3"].lib_paths
-                tools.replace_in_file(
-                    os.path.join(self._source_subfolder, "data", cmake_sqlite_call),
-                    "COMMAND {}".format(pattern),
-                    "COMMAND ${{CMAKE_COMMAND}} -E env \"DYLD_LIBRARY_PATH={}\" {}".format(
-                        ":".join(lib_paths), pattern
-                    ),
-                )
+        if is_apple_os(self):
+            if Version(self.version) < "8.1.0":
+                cmake_sqlite_call = "CMakeLists.txt"
+                pattern = "${EXE_SQLITE3}"
+            else:
+                cmake_sqlite_call = "generate_proj_db.cmake"
+                pattern = "\"${EXE_SQLITE3}\""
+            lib_paths = self.dependencies.build["sqlite3"].cpp_info.libdirs
+            replace_in_file(self,
+                os.path.join(self.source_folder, "data", cmake_sqlite_call),
+                f"COMMAND {pattern}",
+                f"COMMAND ${{CMAKE_COMMAND}} -E env \"DYLD_LIBRARY_PATH={':'.join(lib_paths)}\" {pattern}"
+            )
 
         # unvendor nlohmann_json
-        if tools.Version(self.version) < "8.1.0":
-            tools.rmdir(os.path.join(self._source_subfolder, "include", "proj", "internal", "nlohmann"))
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["USE_THREAD"] = self.options.threadsafe
-        self._cmake.definitions["BUILD_CCT"] = self.options.build_executables
-        self._cmake.definitions["BUILD_CS2CS"] = self.options.build_executables
-        self._cmake.definitions["BUILD_GEOD"] = self.options.build_executables
-        self._cmake.definitions["BUILD_GIE"] = self.options.build_executables
-        self._cmake.definitions["BUILD_PROJ"] = self.options.build_executables
-        self._cmake.definitions["BUILD_PROJINFO"] = self.options.build_executables
-        self._cmake.definitions["PROJ_DATA_SUBDIR"] = "res"
-        if tools.Version(self.version) < "7.0.0":
-            self._cmake.definitions["PROJ_TESTS"] = False
-            self._cmake.definitions["BUILD_LIBPROJ_SHARED"] = self.options.shared
-            self._cmake.definitions["ENABLE_LTO"] = False
-            self._cmake.definitions["JNI_SUPPORT"] = False
+        if Version(self.version) < "8.1.0":
+            rmdir(self, os.path.join(self.source_folder, "include", "proj", "internal", "nlohmann"))
+
+
+    def generate(self):
```

Comment:
move generate() before build() please

Reply:
Ok, I left it in the original order to make the diff easier to read.  Time to move!

Reply:
I feel like _patch_sources() should be folded into build().   Do you like to do that?